locks: remove unused conditional switch to spin_lock code

vh05 opened the pull request:
use of spin_locks is depend on the variable use_spinlocks
but the same is commented in the current code base through
https://review.gluster.org/#/c/glusterfs/+/14763/. So it is
of no use to have conditional switching to spin_lock or
mutex. Removing the dead code as part of the patch

Fixes: #1996
Change-Id: Ib005dd86969ce33d3409164ef3e1011bb3169129
Signed-off-by: Vinayakswami Hariharmath <vharihar@redhat.com>

vh05 committed:
locks: remove ununsed conditional switch to spin_lock code

use of spin_locks is depend on the variable use_spinlocks
but the same is commented in the current code base through
https://review.gluster.org/#/c/glusterfs/+/14763/. So it is
of no use to have conditional switching to spin_lock or
mutex. Removing the dead code as part of the patch

Fixes: #1996
Change-Id: Ib005dd86969ce33d3409164ef3e1011bb3169129
Signed-off-by: Vinayakswami Hariharmath <vharihar@redhat.com>

vh05 commented:
/run regression

vh05 commented:
/run regression

xhernandez commented:
Have you run some performance comparison to be sure this change doesn't cause any regression ?

vh05 commented:
@xhernandez I found that the decider variable use_spinlocks is commented in the codebase and always leading to the use of mutex.

So by default, we are using mutex wherever we used LOCK() and UNLOCK(). I don't think the pthread_spin_lock() part in locking.h is necessary. I am just submitting the patch to remove that part in locking.h

vh05 commented:
/run regression

vh05 commented:
/run regression

vh05 commented:
/recheck smoke

xhernandez reviewed (changes requested):
You should also remove the spinlock detection code from `configure.ac`.

vh05 committed:
locks: remove unused conditional switch to spin_lock code

use of spin_locks is depend on the variable use_spinlocks
but the same is commented in the current code base through
https://review.gluster.org/#/c/glusterfs/+/14763/. So it is
of no use to have conditional switching to spin_lock or
mutex. Removing the dead code as part of the patch

Fixes: #1996
Change-Id: Ib005dd86969ce33d3409164ef3e1011bb3169129
Signed-off-by: Vinayakswami Hariharmath <vharihar@redhat.com>

vh05 commented:
/run regression

xhernandez commented on libglusterfs/src/locking.c:
```diff
@@ -8,7 +8,6 @@
   cases as published by the Free Software Foundation.
 */
 
-#if defined(HAVE_SPINLOCK)
```
Without spinlocks this whole file doesn't make sense. There's nothing to compile. It should be removed.

vh05 committed:
locks: remove unused conditional switch to spin_lock code

use of spin_locks is depend on the variable use_spinlocks
but the same is commented in the current code base through
https://review.gluster.org/#/c/glusterfs/+/14763/. So it is
of no use to have conditional switching to spin_lock or
mutex. Removing the dead code as part of the patch

Fixes: #1996
Change-Id: Ib005dd86969ce33d3409164ef3e1011bb3169129
Signed-off-by: Vinayakswami Hariharmath <vharihar@redhat.com>

vh05 commented:
/run regression

vh05 commented:
/run regression

vh05 commented:
/recheck smoke

vh05 commented:
/recheck smoke
